office365_connector.py

```diff
@@ -486,30 +424,108 @@ def load_state(self):
+            }
+            return state
+        return self._decrypt_state(state, self._asset_id)
 
     def save_state(self, state):
         """
-        Encrypt and save the current state dictionary to the the state file.
+        Encrypt and save the current state dictionary to the state file.
 
         :param state: state dictionary
         :return: status
         """
-        try:
-            state = _encrypt_state(state, self._asset_id)
-        except Exception as e:
-            self._dump_error_log(e)
-            self.debug_print("{}: {}".format(MSGOFFICE365_ENCRYPTION_ERROR, str(e)))
-            return phantom.APP_ERROR
+        return super().save_state(self._encrypt_state(state, self._asset_id))
+
+    def _decrypt_state(self, state, salt):
+        """
+        Decrypts the state.
+
+        :param state: state dictionary
+        :param salt: salt used for decryption
+        :return: decrypted state
+        """
+        if not state.get("is_encrypted"):
+            return state
+
+        if state.get("non_admin_auth", {}).get("access_token"):
+            try:
+                state["non_admin_auth"]["access_token"] = encryption_helper.decrypt(
+                    state["non_admin_auth"]["access_token"], salt
+                )
+            except Exception as ex:
+                self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR,
+                                                              _get_error_msg_from_exception(ex, self)))
+                state["non_admin_auth"]["access_token"] = None
+
+        if state.get("non_admin_auth", {}).get("refresh_token"):
+            try:
+                state["non_admin_auth"]["refresh_token"] = encryption_helper.decrypt(
+                    state["non_admin_auth"]["refresh_token"], salt
+                )
+            except Exception as ex:
+                self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR,
+                                                              _get_error_msg_from_exception(ex, self)))
+                state["non_admin_auth"]["refresh_token"] = None
+
+        if state.get("admin_auth", {}).get("access_token"):
+            try:
+                state["admin_auth"]["access_token"] = encryption_helper.decrypt(
+                    state["admin_auth"]["access_token"], salt
+                )
+            except Exception as ex:
+                self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR,
+                                                              _get_error_msg_from_exception(ex, self)))
+                state["admin_auth"]["access_token"] = None
+
+        return state
+
+    def _encrypt_state(self, state, salt):
+        """
+        Encrypts the state.
+
+        :param state: state dictionary
+        :param salt: salt used for encryption
+        :return: encrypted state
+        """
 
-        return super().save_state(state)
+        if state.get("non_admin_auth", {}).get("access_token"):
```

Comment:
These three blocks are the same except the field being updated, it should be a function that takes the structure to update as argument

Reply:
I made the separate function for updating state variables